ledger/ledger-utxo/src/main/kotlin/net/corda/v5/ledger/utxo/UtxoTransactionBuilder.kt

```diff
@@ -0,0 +1,76 @@
+package net.corda.v5.ledger.utxo
+
+import net.corda.v5.base.annotations.CordaSerializable
+import net.corda.v5.base.annotations.DoNotImplement
+
+/**
+ * Defines a builder for UTXO transactions.
+ */
+@DoNotImplement
+@CordaSerializable
+interface UtxoTransactionBuilder {
+
+    /**
+     * Adds the specified input to the UTXO transaction.
+     *
+     * @param stateRef The state reference of the input to add to the current [UtxoTransactionBuilder].
+     * @return Returns a new [UtxoTransactionBuilder] with the specified input state.
+     */
+    fun withInput(stateRef: StateRef): UtxoTransactionBuilder
+
+    /**
+     * Adds the specified inputs to the UTXO transaction.
+     *
+     * @param stateRefs The state references of the inputs to add to the current [UtxoTransactionBuilder].
+     * @return Returns a new [UtxoTransactionBuilder] with the specified input states.
+     */
+    fun withInputs(stateRefs: Iterable<StateRef>): UtxoTransactionBuilder
```

Comment:
Same here (and throughout) - default vararg funs which in turn call the fun taking an iterable would enable:

```
val myCommands = listOf(command1, command2)
builder.withInputs(input1)
   .withReferenceInputs(refInput1, refInput2)
   .withOutputs(output3)
    .withCommands(myCommands)
```